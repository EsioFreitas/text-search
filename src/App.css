.error-message {
  padding: 1rem;
  background-color: #af2220;
  color: white;
  text-align: center;
  transition: all 1s;
}

.search-container {
  margin: 2rem;
}

.search-header {
  margin-bottom: 2rem;
}

.search-header h1 {
  font-size: 4rem;
}

.search-input {
  display: flex;
  align-items: baseline;
  margin: 1.5rem 0;
  width: 50%;
}

.search-input input {
  margin-right: 1rem;
}

.search-list > h2 {
  font-weight: bold;
  font-size: 2rem;
  margin-bottom: 1.8rem;
}

.search-list > div {
  display: grid;
  grid-template-columns: repeat(auto-fill, 330px);
  gap: 2rem;
}

.image-container {
  display: flex;
  justify-content: center;
}

.void-image {
  width: 35vw;
  margin: 0 auto;
}
