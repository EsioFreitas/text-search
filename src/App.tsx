import React, { useState } from "react";
import "./App.css";
import TextSearchCard from "./components/TextSearchCard";
import { SearchText } from "./models/SearchText";
import api from "./services/Api";

function App() {
  const [searchString, setSearchString] = useState("");
  const [searchTexts, setSearchTexts] = useState<SearchText[]>([]);

  const makeNewSearchText = (searchString: string) => {
    api
      .post("/crawl", { keyword: searchString })
      .then(({ data }) => {
        setSearchTexts([
          ...searchTexts,
          {
            id: data.id,
            status: "active",
            urls: [],
          },
        ]);
      })
      .catch((error) => {});
  };

  return (
    <div className="search-container">
      <header className="search-header">
        <h1>Text Search</h1>
        <div className="search-input">
          <input
            type="text"
            placeholder="Search..."
            value={searchString}
            onChange={(event) => setSearchString(event.target.value)}
          />
          <button onClick={() => makeNewSearchText(searchString)}>
            search
          </button>
        </div>
      </header>
      <main className="search-list">
        <h2>Text search list</h2>
        <div>
          {searchTexts.map((searchText) => (
            <TextSearchCard key={searchText.id} />
          ))}
        </div>
      </main>
    </div>
  );
}

export default App;
