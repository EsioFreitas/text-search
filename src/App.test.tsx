import React from "react";
import { render } from "@testing-library/react";
import App from "./App";

describe("App tests", () => {
  it("should App component return", () => {
    const { baseElement } = render(<App />);
    expect(baseElement).toBeTruthy();
  });

  it("should render App component", async () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
