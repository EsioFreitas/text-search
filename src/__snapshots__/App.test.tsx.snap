// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should render App component 1`] = `
<div>
  <div
    class="backdrop-close"
  >
    <div
      class="modal-close"
    >
      <h3
        class="modal-title"
      />
      <h4
        class="modal-subtitle"
        style="color: rgb(204, 204, 204);"
      />
    </div>
  </div>
  <div
    class="search-container"
  >
    <header
      class="search-header"
    >
      <h1>
        Text Search
      </h1>
      <div
        class="search-input"
      >
        <input
          placeholder="Search..."
          type="text"
          value=""
        />
        <button
          disabled=""
        >
          search
        </button>
      </div>
    </header>
    <main
      class="search-list"
    >
      <h2>
        Text search list
      </h2>
      <div />
    </main>
    <div
      class="image-container"
    >
      <img
        alt="void"
        class="void-image"
        src="void.svg"
      />
    </div>
  </div>
</div>
`;
